### ISSUE TITLE
Bar number class label in SVG output

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#558]
It would be nice to wrap measure/bar numbers in a `g` element with a class name such as `barNum`.

Example of a measure "2":

<img width="166" alt="screen shot 2017-04-23 at 2 07 25 pm" src="https://cloud.githubusercontent.com/assets/3487289/25317481/58f61a98-282e-11e7-9c34-0c1aec7d11d2.png">

The "2" is encoded as a `use` element at the start of the `system` element and before the start of the `g.measure` element for the first measure on the system.

```xml
            <g class="system" id="system-0000000583127447">
                      <use xlink:href="#E882" x="0" y="3620" height="720px" width="720px" />
                      <g class="measure m-2" id="measure-L4">
```

so change to something like:

```xml
            <g class="system" id="system-0000000583127447">
                      <g class="barNum">
			   <use xlink:href="#E882" x="0" y="3620" height="720px" width="720px" />
                      </g>
                      <g class="measure m-2" id="measure-L4">
```

If there are more digits, then each will produce a `use` element:

```xml
	<use xlink:href="#E884" x="0" y="9740" height="720px" width="720px" />
	<use xlink:href="#E884" x="186" y="9740" height="720px" width="720px" />
```

would change to:

```xml
        <g class="barNum">
        	<use xlink:href="#E884" x="0" y="9740" height="720px" width="720px" />
        	<use xlink:href="#E884" x="186" y="9740" height="720px" width="720px" />
       </g>
```

As a sub-issue, it would be better if the bar numbers were not encoded as graphical symbols and rather as ASCII text -- giving the styling of the numbers with a font rather than through a `use` element. (Maybe the `VerovioText` font can do that? Unless the digits slots are used for time signature numbers or similar already).  Otherwise I am wondering how you will deal with a feature request such as a measure labeled "13b" as I think is in the issue list already.

MEI test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-04-23T14:07:19" version="0.9.14-dev-98874c6">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000593602135">
                        <staffGrp xml:id="staffgrp-0000001824208189">
                            <staffDef xml:id="staffdef-0000000813423728" n="1" clef.shape="G" clef.line="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000825956962">
                        <measure xml:id="measure-L2" n="1" type="m-1">
                            <staff xml:id="staff-L2F1N1" n="1">
                                <layer xml:id="layer-L2F1N1" n="1">
                                    <note xml:id="note-L3F1" type="qon-0 qoff-4 pname-c acc-n oct-4 b40c-2 b12c-0 " dur="1" oct="4" pname="c">
                                        <accid xml:id="accid-L3F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L4" n="2" right="end" type="m-2">
                            <staff xml:id="staff-L4F1N1" n="1">
                                <layer xml:id="layer-L4F1N1" n="1">
                                    <note xml:id="note-L5F1" type="qon-4 qoff-8 pname-d acc-n oct-4 b40c-8 b12c-2 " dur="1" oct="4" pname="d">
                                        <accid xml:id="accid-L5F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by rettinghaus** [comment]
I'd like to strongly support these ideas, especially the latter one. Additionally I think this should be inside the `g.measure` element, as it is part of a measure and *not* something *between* two measures.

**Comment by craigsapp** [comment]
Yes, putting it inside the g.measure element is a good idea.

**Comment by pe-ro** [comment]
I have a plan to add an `<mnum>` element in MEI 4.0, for example --
```
<mnum place="above">
  <rend rend="box">32a</rend>
</mnum>
```
This allows more fine-grained control over the visual characteristics of measure numbers, but may cause some confusion over when to use `<measure n="32a">` and when to use `<mnum>`.

Thoughts?

**Comment by craigsapp** [comment]
> Thoughts?

This one popped into my mind: who puts measure numbers in boxes other than mup? :-)

> `<measure n="32a">` and when to use `<mnum>`

One possibility is that `measure@n` could be limited to numbers and simple rendering, and `<mnum>` could be used when special formatting or adding letters to the measure numbers.

In terms of semantics, there are two reasons for adding a letter, so you might want to think about how to encode the structural aspect of the letters rather than the literal letters (and let the rendering software add them according to a house style).

The first method is when a measure is split by a system break, as in this C.P.E. Bach fantasia:

<img width="986" alt="screen shot 2017-04-24 at 8 11 36 pm" src="https://cloud.githubusercontent.com/assets/3487289/25367475/aed1364e-292a-11e7-9e21-f915dc2cbd77.png">

Measure 215 starts on the previous system, and the measure number given at the start of the next system is "215b".  In terms of rendering this letter would be determined by how many times the measure has been split across systems (I have seen about 10 system breaks of a measure in another C.P.E. Bach keyboard work).  Whoever decides the system breaks should get to control the letters in that case: either the human encoder if giving manual system breaks), or the computer if it gets to decide.  In both cases there is not a real need to encode measure letters as they are functional and could be calculated from a more semantic encoding.

The second case comes from the house style of certain publishers.  The first repeat of a measure would be labeled measure 90, while the parallel measure in the second repeat would be called "90a" or "90b" (or not labeled but there would be two measure 90s).  I detest this use of measures letters as there is no point to doing something like that, and it unnecessarily complicates dealing with digital scores.  Here is an example of that system where there are to measure 90s at the end of the example (second one is labeled 90):

<img width="688" alt="screen shot 2017-04-24 at 8 34 46 pm" src="https://cloud.githubusercontent.com/assets/3487289/25367875/8ea1e41a-292d-11e7-9b1c-edc892857e0d.png">

**Comment by pe-ro** [comment]
>[W]ho puts measure numbers in boxes other than mup? :)

Measure "numbers" are boxed when they function as rehearsal "letters". 😀 Just like in the publisher example you provided.

>One possibility is that `measure@n` could be limited to numbers and simple rendering, and `<mnum>` could be used when special formatting or adding letters to the measure numbers.

I agree.  Forcing `@n` to contain only digits (that is, it has to be an integer) reinforces the definition of `@n`: "Provides a numeric designation that indicates an element's position in a sequence of similar elements. Its value must be a non-negative integer."  In other words, if a measure "number" isn't actually a "counting" number, then use `<mNum>`.

It would be confusing and unnecessary, but unfortunately one can alternate between `@n` and `<mNum>`. ☹️  The use of `@n` for all "numbers" was an attempt to reduce the complexity and confusion.  But, making the label an element provides greater control over it graphically; that is, it can have its own placement attributes and the SVG can be better-structured.  So, I think this is the way to go.

>[Y]ou might want to think about how to encode the structural aspect of the letters rather than the literal letters (and let the rendering software add them according to a house style).

I'd prefer to put that kind of info in an attribute attached to `<scoreDef>`, `<staffDef>`, etc. like TEI's `<cRefPattern>`.  It could also be handled by a separate style sheet/language.  I hear there may be work in that area in the future. 😁

**Comment by rettinghaus** [comment]
fixed with #644

**Comment by rettinghaus** [comment]
This can be closed now, I guess.
